numba/tests/test_parallel_backend.py

```diff
@@ -521,6 +522,80 @@ def generate(cls):
 TestThreadingLayerSelection.generate()
 
 
+@skip_parfors_unsupported
+class TestThreadingLayerPriority(ThreadLayerTestHelper):
+
+    def each_env_var(self, env_var: str):
+        """Test setting priority via env var NUMBA_THREADING_LAYER_PRIORITY.
+
+        :return: threading_layer_priority, stderr
+            (containing ``@threading_layer@``)
+        """
+        env = os.environ.copy()
+        env['NUMBA_THREADING_LAYER'] = 'default'
+        env['NUMBA_THREADING_LAYER_PRIORITY'] = env_var
+
+        code = """import sys
+import numba
+
+# trigger threading layer decision
+# hence catching invalid THREADING_LAYER_PRIORITY
+@numba.jit(
+    'float64[::1](float64[::1], float64[::1])',
+    nopython=True,
+    parallel=True,
+)
+def plus(x, y):
+    return x + y
+
+print(' '.join(numba.config.THREADING_LAYER_PRIORITY))
+print("@%s@" % numba.threading_layer(), file=sys.stderr)
```

Comment:
Instead of using stdout/stderr, perhaps just make these into assertions such that the process will exit non-zero if there's a problem. The use of stdout/stderr relies on these streams never having unexpected things written to them (e.g. a deprecation warning) which makes their use more fragile.

Reply:
The stdout should be safe. The stderr pattern `@%s@` is actually copied from other tests in Numba (if you search this string in the same file there are 2 other such uses.) So I was just following whatever is practicing in Numba already...

Reply:
> The stdout should be safe. The stderr pattern `@%s@` is actually copied from other tests in Numba (if you search this string in the same file there are 2 other such uses.) So I was just following whatever is practicing in Numba already...

This is one of the things the code base has been moving away from as it's a little error prone. As it's variables that are ending up as strings to test, having an assert on the actual variables is cheap and reduces maintenance burden. In the interests of getting this feature merged and pre-existing use of this pattern I'm included to leave it. I'll make a note of this pattern as something to refactor generally.

Reply:
https://github.com/numba/numba/issues/7458 tracks.